tracer/src/Datadog.Trace/Ci/Agent/CIWriterHttpSender.cs

```diff
@@ -165,28 +170,55 @@ private async Task SendPayloadAsync<T>(EventPlatformPayload payload, Func<IApiRe
                 }
 
                 Log.Debug<string>("Successfully sent events to {AgentEndpoint}", _apiRequestFactory.Info(url));
-                return;
+                return statusCode;
             }
         }
 
         private async Task SendPayloadAsync(CIVisibilityProtocolPayload payload)
         {
+            var sw = Stopwatch.StartNew();
             var payloadArray = payload.ToArray();
+            TelemetryFactory.Metrics.RecordDistributionCIVisibilityEndpointEventsSerializationMs(payload.TelemetryEndpoint, sw.Elapsed.TotalMilliseconds);
+            TelemetryFactory.Metrics.RecordCountCIVisibilityEndpointPayloadRequests(payload.TelemetryEndpoint);
+            TelemetryFactory.Metrics.RecordDistributionCIVisibilityEndpointPayloadBytes(payload.TelemetryEndpoint, payloadArray.Length);
+            TelemetryFactory.Metrics.RecordDistributionCIVisibilityEndpointPayloadEventsCount(payload.TelemetryEndpoint, payload.Count);
+
+            sw.Reset();
```

Comment:
Shouldn't you use restart? The difference with reset is that it starts measuring time again if i believe the doc 
Same for all the calls to reset in that class.
```suggestion
            sw.Restart();
```

Reply:
Oh, you're absolutely right. Fixed in: https://github.com/DataDog/dd-trace-dotnet/pull/4596/commits/53ec6cfcf5af2063ae7c20389da9cccb58cc6219